pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -0,0 +1,1097 @@
+        assertEq(actualAmountsIn.length, 1, "actualAmountsIn length is wrong");
+        assertEq(actualTokensIn.length, queryTokensIn.length, "actual and query tokensIn lengths do not match");
+        assertEq(actualAmountsIn.length, queryAmountsIn.length, "actual and query amountsIn lengths do not match");
+        assertEq(
+            actualPathAmountsIn.length,
+            queryPathAmountsIn.length,
+            "actual and query pathAmountsIn lengths do not match"
+        );
+
+        assertApproxEqAbs(
+            queryPathAmountsIn[0],
+            actualPathAmountsIn[0],
+            tolerance,
+            "Query and actual pathAmountsIn do not match"
+        );
+        assertEq(queryTokensIn[0], actualTokensIn[0], "Query and actual tokensIn do not match");
+        assertApproxEqAbs(queryAmountsIn[0], actualAmountsIn[0], tolerance, "Query and actual amountsIn do not match");
+
+        _verifySwapResult(actualPathAmountsIn, actualTokensIn, actualAmountsIn, vars);
+    }
+
+    function _verifySwapResult(
+        uint256[] memory paths,
+        address[] memory tokens,
+        uint256[] memory amounts,
+        SwapResultLocals memory vars
+    ) private view {
+        assertEq(paths.length, 1, "Incorrect output array length");
+
+        assertEq(paths.length, tokens.length, "Output array length mismatch");
+        assertEq(tokens.length, amounts.length, "Output array length mismatch");
+
+        // Check results
+        if (vars.kind == SwapKind.EXACT_IN) {
+            // Rounding issues occurs in favor of vault.
+            assertLe(paths[0], vars.expectedDeltaTokenOut, "paths AmountOut must be <= expected amountOut");
+            assertLe(paths[0], vars.expectedDeltaTokenOut, "amounts AmountOut must be <= expected amountOut");
+
+            // Rounding issues are very small.
+            assertApproxEqAbs(paths[0], vars.expectedDeltaTokenOut, 2, "Wrong path count");
+            assertApproxEqAbs(amounts[0], vars.expectedDeltaTokenOut, 2, "Wrong amounts count");
+            assertEq(tokens[0], address(vars.tokenOut), "Wrong token for SwapKind");
+        } else {
+            // Rounding issues occurs in favor of vault.
+            assertGe(paths[0], vars.expectedDeltaTokenIn, "paths AmountIn must be >= expected amountIn");
+            assertGe(amounts[0], vars.expectedDeltaTokenIn, "amounts AmountIn must be >= expected amountIn");
+
+            // Rounding issues are very small.
+            assertApproxEqAbs(paths[0], vars.expectedDeltaTokenIn, 5, "Wrong path count");
+            assertApproxEqAbs(amounts[0], vars.expectedDeltaTokenIn, 5, "Wrong amounts count");
+            assertEq(tokens[0], address(vars.tokenIn), "Wrong token for SwapKind");
+        }
+
+        // Tokens were transferred
+        assertLe(
+            vars.tokenIn.balanceOf(lp),
+            vars.lpBeforeSwapTokenIn - vars.expectedDeltaTokenIn,
+            "LP balance tokenIn must be <= expected balance"
```

Comment:
Can we clarify that this `Le` is to test the rounding in favor of the Vault with a comment?